### PR TITLE
search: add test via GraphQL queries

### PR DESCRIPTION
**Pull request by ijt**
This should make it easier for us to write clear tests that work regardless of how search is implemented.

This PR just adds the empty case so we get the structure in place. Later changes will add additional test cases.

### CODE REVIEW

**Comment by ijt** [comment]
CC @nicksnyder , @ijsnow, @slimsag , @attfarhan

**Comment by ijt** [comment]
Keegan isn't working today. @slimsag, can you review this?

File: cmd/frontend/graphqlbackend/search_test.go [hunk 1]
```diff
@@ -1,5 +1,181 @@
...
+				t.Fatalf("graphQL query returned errors: %+v", result.Errors)
+			}
+			var search struct {
+				Results Results
+			}
+			if err := json.Unmarshal(result.Data, &search); err != nil {
+				t.Fatalf("parsing JSON response: %v", err)
+			}
+			gotResults := search.Results
+			if !reflect.DeepEqual(gotResults, tc.wantResults) {
```
Comment (ijsnow) [inline cmd/frontend/graphqlbackend/search_test.go]: This is will only work if we only write tests with less than 1000 results.

Comment (ijt) [reply cmd/frontend/graphqlbackend/search_test.go]: I don't see any such restriction in `DeepEqual`. https://golang.org/src/reflect/deepequal.go

Can you explain?

Comment (ijsnow) [reply cmd/frontend/graphqlbackend/search_test.go]: It looks like it depends on a stable order but we don’t stable sort result sets that are greater than 1000

Comment (nicksnyder) [reply cmd/frontend/graphqlbackend/search_test.go]: @ijsnow where is the code for that? It is something that we discussed, but last time I checked we sort everything that we get back.

**Pull request merged**